### ISSUE TITLE
Automate `org.yaml` generation

### ISSUE THREAD

**Issue by zuc** [falcosecurity/evolution#195]
**Motivation**

Current peribolos file is being manually curated. This is of course prone to human error.

**Feature**

Generate `org.yaml` automatically.

**Alternatives**

N/A

**Additional context**

This should be done after #176 is completed.

**Comment by zuc** [comment]
Note to self: current maintainers/repositories yaml files purposely exclude some data that is actually needed by peribolos (eg. archived repos).

**Comment by leogr** [comment]
> Note to self: current maintainers/repositories yaml files purposely exclude some data that is actually needed by peribolos (eg. archived repos).

AFAIK, one must first archive the repository using the GH UI. Otherwise `archived: true` in peribolos won't work. Should we get that information from the GH API? :thinking: 

Also, the only path to invite a new member to the org is by adding it to `org.yaml`. I guess that will remain a change that has to be manually applied to `org.yaml`, right?

**Comment by zuc** [comment]
Fetching repo info from GH APIs to feed a file that is supposed to manage/update the same repo(s) doesn't seem a good path to me.

I still have to draw a plan on this but what I'd like to do is:
1. Have a single source of truth about our repos in a general purpose, machine-readable format (that could be the existing https://github.com/falcosecurity/evolution/blob/main/repositories.yaml file which needs to be enriched with some data. This file will be manually curated to represent the desired state of the repos org-wide.
2. For each repo listed in the above-mentioned file, fetch OWNERS file content; that could be done either using https://github.com/leodido/maintainers-generator or through a custom util.
3. Have all the automation processes consume those files to perform all other ops (eg. generate `org.yaml`, generate markdown(s), ...).

**Comment by zuc** [comment]
cc @jasondellaluce as he curated current automations

**Comment by leogr** [comment]
Somehow related, IMO :point_down: 
https://github.com/falcosecurity/evolution/issues/198

**Comment by leogr** [comment]
/remove-lifecycle stale

we still want this

**Comment by maxgio92** [comment]
I have a question about this: why not org.yaml Peribolos config could be the source of truth about repositories too?

**Comment by maxgio92** [comment]
Sure, thank you @leogr.

**Comment by leogr** [comment]
We need to rethink this since the CNCF introduced https://github.com/falcosecurity/.project
In particular, I want to avoid redundancies and keep the data from being spread across many places. 

I'm closing this for now, and I will open a new issue once I find the time to analyze the whole situation again.

/close
